lnrpc/walletrpc/walletkit_server.go

```diff
@@ -645,6 +649,39 @@ func (w *WalletKit) PublishTransaction(ctx context.Context,
 	return &PublishResponse{}, nil
 }
 
+// Attempts to remove the transaction and all of its descendants resulting from
+// further spends of the outputs of the provided transaction.
+// NOTE: We do not remove the transaction from the rebroadcaster which might
+// run in the background rebroadcasting not yet confirmed transactions. We do
+// not have access to the rebroadcaster here nor should we. This command is not
+// a way to remove transactions from the network. It is a way to shortcircuit
+// wallet utxo housekeeping while transactions are still unconfirmed and we know
+// that a transaction will never confirm because a replacement already pays
+// higher fees.
+func (w *WalletKit) RemoveTransaction(_ context.Context,
+	req *Transaction) (*RemoveTransactionResponse, error) {
+
+	// If the client doesn't specify a transaction, then there's nothing to
+	// be removed.
+	if len(req.TxHex) == 0 {
+		return nil, fmt.Errorf("must provide a transaction to " +
+			"remove")
+	}
+
+	tx := &wire.MsgTx{}
+	txReader := bytes.NewReader(req.TxHex)
+	if err := tx.Deserialize(txReader); err != nil {
+		return nil, err
+	}
+
+	err := w.cfg.Wallet.RemoveDescendants(tx)
```

Comment:
I wonder: if instead of requiring the user to provide the full serialized transaction, couldn't we instead just ask for the TXID, then look it up in the wallet? If we don't have it in the wallet, then we also don't need to remove it, right?

Reply:
https://github.com/btcsuite/btcwallet/pull/779 might be useful for that.

Reply:
cool, makes sense.

Reply:
reused the `ListTransactionDetails` fn wdyt ?

Reply:
Nice! I guess we can simplify this code a bit once https://github.com/btcsuite/btcwallet/pull/779 is in, but that's alright for now.